numpy/core/src/multiarray/buffer.c

```diff
@@ -266,8 +266,11 @@ _buffer_format_string(PyArray_Descr *descr, _tmp_string_t *str,
             }
 
             /* Insert child item */
-            _buffer_format_string(child, str, obj, offset,
+            ret = _buffer_format_string(child, str, obj, offset,
                                   active_byteorder);
+            if (ret != 0) {
+                return ret;
```

Comment:
I'd prefer `-1` here - this gives the impression that we actually have multiple types of error code, which is not the case

Reply:
Fixing

It is a shame we cannot use the sentinels for success and failure that we define in our headers:
```
#define NPY_FAIL 0
#define NPY_SUCCEED 1
```

Reply:
The numpy versions are non-standard and weird. They are appropriate for bash scripts, which is why they are used in `int main functions`, but a more standard unix convention is negative numbers for errors, in particular, -1 is popular. Even standardized positive error values are returned as their negative and converted to positive for display.  Oddly, the topic is rarely covered in style guides.

Reply:
`NPY_FAIL` and `NPY_SUCCEED` are only for use in `PyArg_ParseTuple`, which is determined by the CPython API.